src/lib/y2network/interface_config_builder.rb

```diff
@@ -343,13 +343,16 @@ def hostname=(value)
       @hostname = value
     end
 
+    # Saves the hostname
+    #
+    # It needs to take into account whether the old configuration and the boot protocol.
```

Comment:
this sentence is hard to understand for me. Maybe mention that it needs to clear old hostname entry and set a new one even when ip and hostname changes.